diff --git a/src/gui/BasicComboBox.qml b/src/gui/BasicComboBox.qml
index f24c04818..1c2ba78b8 100644
--- a/src/gui/BasicComboBox.qml
+++ b/src/gui/BasicComboBox.qml
@@ -12,10 +12,10 @@
  * for more details.
  */
 
-import QtQuick 2.15
-import QtQuick.Dialogs 1.3
-import QtQuick.Layouts 1.15
-import QtQuick.Controls 2.15
+import QtQuick 2.12
+import QtQuick.Dialogs 1.2
+import QtQuick.Layouts 1.12
+import QtQuick.Controls 2.5
 import QtGraphicalEffects 1.0
 
 import Style 1.0
diff --git a/src/gui/EmojiPicker.qml b/src/gui/EmojiPicker.qml
index 8a9cc920e..876add095 100644
--- a/src/gui/EmojiPicker.qml
+++ b/src/gui/EmojiPicker.qml
@@ -12,9 +12,9 @@
  * for more details.
  */
 
-import QtQuick 2.15
-import QtQuick.Controls 2.15
-import QtQuick.Layouts 1.15
+import QtQuick 2.12
+import QtQuick.Controls 2.5
+import QtQuick.Layouts 1.12
 
 import Style 1.0
 import com.nextcloud.desktopclient 1.0 as NC
diff --git a/src/gui/ErrorBox.qml b/src/gui/ErrorBox.qml
index 750a4cb00..4b072889f 100644
--- a/src/gui/ErrorBox.qml
+++ b/src/gui/ErrorBox.qml
@@ -1,5 +1,5 @@
-import QtQuick 2.15
-import QtQuick.Controls 2.15
+import QtQuick 2.12
+import QtQuick.Controls 2.5
 
 import Style 1.0
 
diff --git a/src/gui/PredefinedStatusButton.qml b/src/gui/PredefinedStatusButton.qml
index dabef84ae..6ce460df3 100644
--- a/src/gui/PredefinedStatusButton.qml
+++ b/src/gui/PredefinedStatusButton.qml
@@ -12,10 +12,10 @@
  * for more details.
  */
 
-import QtQuick 2.15
-import QtQuick.Dialogs 1.3
-import QtQuick.Layouts 1.15
-import QtQuick.Controls 2.15
+import QtQuick 2.12
+import QtQuick.Dialogs 1.2
+import QtQuick.Layouts 1.12
+import QtQuick.Controls 2.5
 
 import Style 1.0
 
diff --git a/src/gui/UserStatusSelector.qml b/src/gui/UserStatusSelector.qml
index eebcee586..5bfb43720 100644
--- a/src/gui/UserStatusSelector.qml
+++ b/src/gui/UserStatusSelector.qml
@@ -13,10 +13,10 @@
  */
 
 import QtQuick 2.6
-import QtQuick.Dialogs 1.3
-import QtQuick.Layouts 1.15
-import QtQuick.Controls 2.15
-import QtQuick.Window 2.15
+import QtQuick.Dialogs 1.2
+import QtQuick.Layouts 1.12
+import QtQuick.Controls 2.5
+import QtQuick.Window 2.12
 import com.nextcloud.desktopclient 1.0 as NC
 
 import Style 1.0
@@ -303,7 +303,6 @@ ColumnLayout {
 
                     model: userStatusSelectorModel.clearStageTypes
                     textRole: "display"
-                    valueRole: "clearStageType"
                     displayText: userStatusSelectorModel.clearAtDisplayString
                     onActivated: userStatusSelectorModel.setClearAt(currentValue)
                 }
diff --git a/src/gui/UserStatusSelectorButton.qml b/src/gui/UserStatusSelectorButton.qml
index 8321232df..0738530e5 100644
--- a/src/gui/UserStatusSelectorButton.qml
+++ b/src/gui/UserStatusSelectorButton.qml
@@ -13,9 +13,9 @@
  */
 
 import QtQuick 2.6
-import QtQuick.Dialogs 1.3
-import QtQuick.Layouts 1.15
-import QtQuick.Controls 2.15
+import QtQuick.Dialogs 1.2
+import QtQuick.Layouts 1.12
+import QtQuick.Controls 2.5
 
 import Style 1.0
 
diff --git a/src/gui/UserStatusSelectorPage.qml b/src/gui/UserStatusSelectorPage.qml
index c89d821d9..a15d42cb9 100644
--- a/src/gui/UserStatusSelectorPage.qml
+++ b/src/gui/UserStatusSelectorPage.qml
@@ -12,8 +12,8 @@
  * for more details.
  */
 
-import QtQuick 2.15
-import QtQuick.Controls 2.15
+import QtQuick 2.12
+import QtQuick.Controls 2.12
 import Style 1.0
 
 import com.nextcloud.desktopclient 1.0 as NC
diff --git a/src/gui/systray.cpp b/src/gui/systray.cpp
index 51ba0e267..3adfb74db 100644
--- a/src/gui/systray.cpp
+++ b/src/gui/systray.cpp
@@ -256,13 +256,13 @@ void Systray::createCallDialog(const Activity &callNotification, const AccountSt
             });
         }
 
-        const QVariantMap initialProperties{
-            {"accountState", QVariant::fromValue(accountState.data())},
-            {"talkNotificationData", talkNotificationData},
-            {"links", links},
-            {"subject", callNotification._subject},
-            {"link", callNotification._link},
-        };
+        // const QVariantMap initialProperties{
+        //     {"accountState", QVariant::fromValue(accountState.data())},
+        //     {"talkNotificationData", talkNotificationData},
+        //     {"links", links},
+        //     {"subject", callNotification._subject},
+        //     {"link", callNotification._link},
+        // };
 
         const auto callDialog = new QQmlComponent(_trayEngine, QStringLiteral("qrc:/qml/src/gui/tray/CallNotificationDialog.qml"));
 
@@ -271,7 +271,14 @@ void Systray::createCallDialog(const Activity &callNotification, const AccountSt
             return;
         }
 
-        callDialog->createWithInitialProperties(initialProperties);
+        auto component = callDialog->beginCreate(_trayEngine->rootContext());
+        callDialog->setProperty("accountState", QVariant::fromValue(accountState.data()));
+        callDialog->setProperty("talkNotificationData", talkNotificationData);
+        callDialog->setProperty("links", links);
+        callDialog->setProperty("subject", callNotification._subject);
+        callDialog->setProperty("link", callNotification._link);
+        callDialog->completeCreate();
+        
         _callsAlreadyNotified.insert(callNotification._id);
     }
 }
diff --git a/src/gui/tray/ActivityActionButton.qml b/src/gui/tray/ActivityActionButton.qml
index eca55ebdc..9f37dbbc4 100644
--- a/src/gui/tray/ActivityActionButton.qml
+++ b/src/gui/tray/ActivityActionButton.qml
@@ -1,6 +1,6 @@
-import QtQuick 2.15
+import QtQuick 2.12
 import QtQuick.Controls 2.3
-import QtQuick.Layouts 1.15
+import QtQuick.Layouts 1.12
 import Style 1.0
 import com.nextcloud.desktopclient 1.0
 
diff --git a/src/gui/tray/ActivityItem.qml b/src/gui/tray/ActivityItem.qml
index b0ead37ed..2bfdd2688 100644
--- a/src/gui/tray/ActivityItem.qml
+++ b/src/gui/tray/ActivityItem.qml
@@ -1,7 +1,7 @@
-import QtQml 2.15
-import QtQuick 2.15
-import QtQuick.Controls 2.15
-import QtQuick.Layouts 1.15
+import QtQml 2.12
+import QtQuick 2.12
+import QtQuick.Controls 2.5
+import QtQuick.Layouts 1.12
 import Style 1.0
 import com.nextcloud.desktopclient 1.0
 
diff --git a/src/gui/tray/ActivityItemActions.qml b/src/gui/tray/ActivityItemActions.qml
index 1b1b70e5c..efce73804 100644
--- a/src/gui/tray/ActivityItemActions.qml
+++ b/src/gui/tray/ActivityItemActions.qml
@@ -1,5 +1,5 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import QtQuick.Layouts 1.2
 import Style 1.0
@@ -102,6 +102,7 @@ RowLayout {
                 function onMovementStarted() {
                     moreActionsButtonContextMenu.close();
                 }
+                onMovementStarted: onMovementStarted()
             }
 
             ActivityItemContextMenu {
diff --git a/src/gui/tray/ActivityItemContent.qml b/src/gui/tray/ActivityItemContent.qml
index 5b4e14fad..83755c805 100644
--- a/src/gui/tray/ActivityItemContent.qml
+++ b/src/gui/tray/ActivityItemContent.qml
@@ -1,9 +1,9 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import QtQuick.Layouts 1.2
 import Style 1.0
-import QtGraphicalEffects 1.15
+import QtGraphicalEffects 1.12
 import com.nextcloud.desktopclient 1.0
 
 RowLayout {
diff --git a/src/gui/tray/ActivityItemContextMenu.qml b/src/gui/tray/ActivityItemContextMenu.qml
index de35e3584..ad150bc0b 100644
--- a/src/gui/tray/ActivityItemContextMenu.qml
+++ b/src/gui/tray/ActivityItemContextMenu.qml
@@ -1,5 +1,5 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import Style 1.0
 
diff --git a/src/gui/tray/ActivityList.qml b/src/gui/tray/ActivityList.qml
index f3a6416c2..c68154c03 100644
--- a/src/gui/tray/ActivityList.qml
+++ b/src/gui/tray/ActivityList.qml
@@ -1,5 +1,5 @@
-import QtQuick 2.15
-import QtQuick.Controls 2.15
+import QtQuick 2.12
+import QtQuick.Controls 2.5
 
 import com.nextcloud.desktopclient 1.0 as NC
 
diff --git a/src/gui/tray/AutoSizingMenu.qml b/src/gui/tray/AutoSizingMenu.qml
index efc23d1bf..3c750eaa1 100644
--- a/src/gui/tray/AutoSizingMenu.qml
+++ b/src/gui/tray/AutoSizingMenu.qml
@@ -1,4 +1,4 @@
-import QtQuick 2.15
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import Style 1.0
 
diff --git a/src/gui/tray/CallNotificationDialog.qml b/src/gui/tray/CallNotificationDialog.qml
index 23d9e2bc8..b8933670e 100644
--- a/src/gui/tray/CallNotificationDialog.qml
+++ b/src/gui/tray/CallNotificationDialog.qml
@@ -13,14 +13,14 @@
  * for more details.
  */
 
-import QtQuick 2.15
-import QtQuick.Window 2.15
+import QtQuick 2.12
+import QtQuick.Window 2.12
 import Style 1.0
 import com.nextcloud.desktopclient 1.0
 import QtQuick.Layouts 1.2
-import QtMultimedia 5.15
-import QtQuick.Controls 2.15
-import QtGraphicalEffects 1.15
+import QtMultimedia 5.12
+import QtQuick.Controls 2.5
+import QtGraphicalEffects 1.0
 
 Window {
     id: root
diff --git a/src/gui/tray/CustomButton.qml b/src/gui/tray/CustomButton.qml
index 32d6ff575..fb7c21790 100644
--- a/src/gui/tray/CustomButton.qml
+++ b/src/gui/tray/CustomButton.qml
@@ -1,4 +1,4 @@
-import QtQuick 2.15
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import QtQuick.Layouts 1.2
 import Style 1.0
diff --git a/src/gui/tray/CustomTextButton.qml b/src/gui/tray/CustomTextButton.qml
index 000a54d3c..d8b32eb43 100644
--- a/src/gui/tray/CustomTextButton.qml
+++ b/src/gui/tray/CustomTextButton.qml
@@ -1,4 +1,4 @@
-import QtQuick 2.15
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import Style 1.0
 
diff --git a/src/gui/tray/FileActivityDialog.qml b/src/gui/tray/FileActivityDialog.qml
index b2d5fb809..66029cf20 100644
--- a/src/gui/tray/FileActivityDialog.qml
+++ b/src/gui/tray/FileActivityDialog.qml
@@ -1,6 +1,6 @@
-import QtQml 2.15
-import QtQuick 2.15
-import QtQuick.Window 2.15
+import QtQml 2.12
+import QtQuick 2.12
+import QtQuick.Window 2.12
 
 import Style 1.0
 import com.nextcloud.desktopclient 1.0 as NC
diff --git a/src/gui/tray/NCBusyIndicator.qml b/src/gui/tray/NCBusyIndicator.qml
index c489e5f39..cc963a9ee 100644
--- a/src/gui/tray/NCBusyIndicator.qml
+++ b/src/gui/tray/NCBusyIndicator.qml
@@ -12,8 +12,8 @@
  * for more details.
  */
 
-import QtQuick 2.15
-import QtQuick.Controls 2.15
+import QtQuick 2.12
+import QtQuick.Controls 2.5
 import Style 1.0
 
 BusyIndicator {
diff --git a/src/gui/tray/SyncStatus.qml b/src/gui/tray/SyncStatus.qml
index bae630a4d..6babceb9f 100644
--- a/src/gui/tray/SyncStatus.qml
+++ b/src/gui/tray/SyncStatus.qml
@@ -1,6 +1,6 @@
-import QtQuick 2.15
-import QtQuick.Controls 2.15
-import QtQuick.Layouts 1.15
+import QtQuick 2.12
+import QtQuick.Controls 2.5
+import QtQuick.Layouts 1.12
 
 import Style 1.0
 
diff --git a/src/gui/tray/TalkReplyTextField.qml b/src/gui/tray/TalkReplyTextField.qml
index bec9e11df..a2a9deccb 100644
--- a/src/gui/tray/TalkReplyTextField.qml
+++ b/src/gui/tray/TalkReplyTextField.qml
@@ -1,6 +1,6 @@
-import QtQuick 2.15
-import QtQuick.Controls 2.15
-import QtQuick.Layouts 1.15
+import QtQuick 2.12
+import QtQuick.Controls 2.5
+import QtQuick.Layouts 1.12
 
 import com.nextcloud.desktopclient 1.0
 import Style 1.0
diff --git a/src/gui/tray/UnifiedSearchInputContainer.qml b/src/gui/tray/UnifiedSearchInputContainer.qml
index 6dc0e17a7..543951a0b 100644
--- a/src/gui/tray/UnifiedSearchInputContainer.qml
+++ b/src/gui/tray/UnifiedSearchInputContainer.qml
@@ -1,5 +1,5 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import QtGraphicalEffects 1.0
 import Style 1.0
diff --git a/src/gui/tray/UnifiedSearchResultFetchMoreTrigger.qml b/src/gui/tray/UnifiedSearchResultFetchMoreTrigger.qml
index 83982841d..04503ffba 100644
--- a/src/gui/tray/UnifiedSearchResultFetchMoreTrigger.qml
+++ b/src/gui/tray/UnifiedSearchResultFetchMoreTrigger.qml
@@ -1,5 +1,5 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import QtQuick.Layouts 1.2
 import Style 1.0
diff --git a/src/gui/tray/UnifiedSearchResultItem.qml b/src/gui/tray/UnifiedSearchResultItem.qml
index e6e7d1fcc..ae4661d2a 100644
--- a/src/gui/tray/UnifiedSearchResultItem.qml
+++ b/src/gui/tray/UnifiedSearchResultItem.qml
@@ -1,4 +1,4 @@
-import QtQml 2.15
+import QtQml 2.12
 import QtQuick 2.9
 import QtQuick.Controls 2.3
 import QtQuick.Layouts 1.2
diff --git a/src/gui/tray/UnifiedSearchResultItemSkeleton.qml b/src/gui/tray/UnifiedSearchResultItemSkeleton.qml
index 32e19cca3..6cbf7c5f0 100644
--- a/src/gui/tray/UnifiedSearchResultItemSkeleton.qml
+++ b/src/gui/tray/UnifiedSearchResultItemSkeleton.qml
@@ -1,7 +1,7 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import QtQuick.Layouts 1.2
-import QtGraphicalEffects 1.15
+import QtGraphicalEffects 1.0
 import Style 1.0
 
 RowLayout {
diff --git a/src/gui/tray/UnifiedSearchResultItemSkeletonContainer.qml b/src/gui/tray/UnifiedSearchResultItemSkeletonContainer.qml
index 354835d51..935a082ea 100644
--- a/src/gui/tray/UnifiedSearchResultItemSkeletonContainer.qml
+++ b/src/gui/tray/UnifiedSearchResultItemSkeletonContainer.qml
@@ -1,5 +1,5 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import Style 1.0
 
 Column {
diff --git a/src/gui/tray/UnifiedSearchResultListItem.qml b/src/gui/tray/UnifiedSearchResultListItem.qml
index c53ca4066..562a11368 100644
--- a/src/gui/tray/UnifiedSearchResultListItem.qml
+++ b/src/gui/tray/UnifiedSearchResultListItem.qml
@@ -1,5 +1,5 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import Style 1.0
 import com.nextcloud.desktopclient 1.0
diff --git a/src/gui/tray/UnifiedSearchResultNothingFound.qml b/src/gui/tray/UnifiedSearchResultNothingFound.qml
index cee73c336..1d41a07dd 100644
--- a/src/gui/tray/UnifiedSearchResultNothingFound.qml
+++ b/src/gui/tray/UnifiedSearchResultNothingFound.qml
@@ -1,5 +1,5 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import QtQuick.Layouts 1.2
 import Style 1.0
@@ -7,7 +7,7 @@ import Style 1.0
 ColumnLayout {
     id: unifiedSearchResultNothingFoundContainer
 
-    required property string text
+    property string text
 
     spacing: Style.standardSpacing
     anchors.leftMargin: Style.unifiedSearchResultNothingFoundHorizontalMargin
diff --git a/src/gui/tray/UnifiedSearchResultSectionItem.qml b/src/gui/tray/UnifiedSearchResultSectionItem.qml
index ebc691a9c..4e91f6f56 100644
--- a/src/gui/tray/UnifiedSearchResultSectionItem.qml
+++ b/src/gui/tray/UnifiedSearchResultSectionItem.qml
@@ -1,12 +1,12 @@
-import QtQml 2.15
-import QtQuick 2.15
-import QtQuick.Controls 2.15
+import QtQml 2.12
+import QtQuick 2.12
+import QtQuick.Controls 2.5
 import QtQuick.Layouts 1.2
 import Style 1.0
 import com.nextcloud.desktopclient 1.0
 
 Label {
-    required property string section
+    property string section
 
     topPadding: Style.unifiedSearchResultSectionItemVerticalPadding
     bottomPadding: Style.unifiedSearchResultSectionItemVerticalPadding
diff --git a/src/gui/tray/Window.qml b/src/gui/tray/Window.qml
index 867bfc5c4..214e838ea 100644
--- a/src/gui/tray/Window.qml
+++ b/src/gui/tray/Window.qml
@@ -1,9 +1,9 @@
-import QtQuick 2.15
+import QtQuick 2.12
 import QtQuick.Window 2.3
 import QtQuick.Controls 2.3
 import QtQuick.Layouts 1.2
 import QtGraphicalEffects 1.0
-import Qt.labs.platform 1.1 as NativeDialogs
+import Qt.labs.platform 1.0 as NativeDialogs
 import "../"
 
 // Custom qml modules are in /theme (and included by resources.qrc)
@@ -64,6 +64,7 @@ ApplicationWindow {
             accountMenu.close();
             syncStatus.model.load();
         }
+        onNewUserSelected: onNewUserSelected()
     }
 
     Component {
@@ -90,16 +91,19 @@ ApplicationWindow {
                 appsMenu.close();
             }
         }
+        onIsOpenChanged: onIsOpenChanged()
 
         function onShowFileActivityDialog(objectName, objectId) {
             openFileActivityDialog(objectName, objectId)
         }
+        onShowFileActivityDialog: onShowFileActivityDialog(sharePath, localPath)
 
         function onShowErrorMessageDialog(error) {
             var newErrorDialog = errorMessageDialog.createObject(trayWindow)
             newErrorDialog.text = error
             newErrorDialog.open()
         }
+        onShowErrorMessageDialog: onShowErrorMessageDialog(error)
     }
 
     OpacityMask {
@@ -781,7 +785,7 @@ ApplicationWindow {
 
                 keyNavigationEnabled: true
 
-                reuseItems: true
+                //TODO: enable it when Qt 5.15 is available for every OS reuseItems: true
 
                 Accessible.role: Accessible.List
                 Accessible.name: qsTr("Unified search results list")
@@ -794,8 +798,6 @@ ApplicationWindow {
                     currentFetchMoreInProgressProviderId: unifiedSearchResultsListView.model.currentFetchMoreInProgressProviderId
                     fetchMoreTriggerClicked: unifiedSearchResultsListView.model.fetchMoreTriggerClicked
                     resultClicked: unifiedSearchResultsListView.model.resultClicked
-                    ListView.onPooled: isPooled = true
-                    ListView.onReused: isPooled = false
                 }
 
                 section.property: "providerName"
diff --git a/theme/Style/Style.qml b/theme/Style/Style.qml
index 6d1adb5ec..af38afdf2 100644
--- a/theme/Style/Style.qml
+++ b/theme/Style/Style.qml
@@ -1,6 +1,6 @@
 pragma Singleton
 
-import QtQuick 2.15
+import QtQuick 2.12
 
 import com.nextcloud.desktopclient 1.0
 
