diff --git a/src/gui/EmojiPicker.qml b/src/gui/EmojiPicker.qml
index 0e5c6ec70..2e6c20e33 100644
--- a/src/gui/EmojiPicker.qml
+++ b/src/gui/EmojiPicker.qml
@@ -12,9 +12,9 @@
  * for more details.
  */
 
-import QtQuick 2.15
-import QtQuick.Controls 2.15
-import QtQuick.Layouts 1.15
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.12
 
 import Style 1.0
 import com.nextcloud.desktopclient 1.0 as NC
diff --git a/src/gui/ErrorBox.qml b/src/gui/ErrorBox.qml
index 750a4cb00..204713d89 100644
--- a/src/gui/ErrorBox.qml
+++ b/src/gui/ErrorBox.qml
@@ -1,5 +1,5 @@
-import QtQuick 2.15
-import QtQuick.Controls 2.15
+import QtQuick 2.12
+import QtQuick.Controls 2.12
 
 import Style 1.0
 
diff --git a/src/gui/UserStatusSelector.qml b/src/gui/UserStatusSelector.qml
index 05f092151..5dfc58d49 100644
--- a/src/gui/UserStatusSelector.qml
+++ b/src/gui/UserStatusSelector.qml
@@ -14,9 +14,9 @@
 
 import QtQuick 2.6
 import QtQuick.Dialogs 1.3
-import QtQuick.Layouts 1.15
-import QtQuick.Controls 2.15
-import QtQuick.Window 2.15
+import QtQuick.Layouts 1.12
+import QtQuick.Controls 2.12
+import QtQuick.Window 2.12
 import com.nextcloud.desktopclient 1.0 as NC
 
 import Style 1.0
diff --git a/src/gui/UserStatusSelectorDialog.qml b/src/gui/UserStatusSelectorDialog.qml
index 938eae547..f6aa4d287 100644
--- a/src/gui/UserStatusSelectorDialog.qml
+++ b/src/gui/UserStatusSelectorDialog.qml
@@ -1,4 +1,4 @@
-import QtQuick.Window 2.15
+import QtQuick.Window 2.12
 import Style 1.0
 
 import com.nextcloud.desktopclient 1.0 as NC
diff --git a/src/gui/systray.cpp b/src/gui/systray.cpp
index 9db7ef835..2534c20b2 100644
--- a/src/gui/systray.cpp
+++ b/src/gui/systray.cpp
@@ -219,16 +219,22 @@ void Systray::createCallDialog(const Activity &callNotification, const AccountSt
             });
         }
 
-        const QVariantMap initialProperties{
-            {"accountState", QVariant::fromValue(accountState.data())},
-            {"talkNotificationData", talkNotificationData},
-            {"links", links},
-            {"subject", callNotification._subject},
-            {"link", callNotification._link},
-        };
+        // const QVariantMap initialProperties{
+        //     {"accountState", QVariant::fromValue(accountState.data())},
+        //     {"talkNotificationData", talkNotificationData},
+        //     {"links", links},
+        //     {"subject", callNotification._subject},
+        //     {"link", callNotification._link},
+        // };
 
         const auto callDialog = new QQmlComponent(_trayEngine, QStringLiteral("qrc:/qml/src/gui/tray/CallNotificationDialog.qml"));
-        callDialog->createWithInitialProperties(initialProperties);
+        auto component = callDialog->beginCreate(_trayEngine->rootContext());
+        callDialog->setProperty("accountState", QVariant::fromValue(accountState.data()));
+        callDialog->setProperty("talkNotificationData", talkNotificationData);
+        callDialog->setProperty("links", links);
+        callDialog->setProperty("subject", callNotification._subject);
+        callDialog->setProperty("link", callNotification._link);
+        callDialog->completeCreate();
 
         _callsAlreadyNotified.insert(callNotification._id);
     }
diff --git a/src/gui/tray/ActivityItemActions.qml b/src/gui/tray/ActivityItemActions.qml
index 1b1b70e5c..d4a90ee79 100644
--- a/src/gui/tray/ActivityItemActions.qml
+++ b/src/gui/tray/ActivityItemActions.qml
@@ -102,6 +102,7 @@ RowLayout {
                 function onMovementStarted() {
                     moreActionsButtonContextMenu.close();
                 }
+                onMovementStarted: onMovementStarted()
             }
 
             ActivityItemContextMenu {
diff --git a/src/gui/tray/ActivityList.qml b/src/gui/tray/ActivityList.qml
index acfb1bbdc..1d4af0300 100644
--- a/src/gui/tray/ActivityList.qml
+++ b/src/gui/tray/ActivityList.qml
@@ -1,5 +1,5 @@
-import QtQuick 2.15
-import QtQuick.Controls 2.15
+import QtQuick 2.12
+import QtQuick.Controls 2.12
 
 import com.nextcloud.desktopclient 1.0 as NC
 
diff --git a/src/gui/tray/AutoSizingMenu.qml b/src/gui/tray/AutoSizingMenu.qml
index efc23d1bf..3c750eaa1 100644
--- a/src/gui/tray/AutoSizingMenu.qml
+++ b/src/gui/tray/AutoSizingMenu.qml
@@ -1,4 +1,4 @@
-import QtQuick 2.15
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import Style 1.0
 
diff --git a/src/gui/tray/FileActivityDialog.qml b/src/gui/tray/FileActivityDialog.qml
index 58264ef28..61905bcbf 100644
--- a/src/gui/tray/FileActivityDialog.qml
+++ b/src/gui/tray/FileActivityDialog.qml
@@ -1,6 +1,6 @@
-import QtQml 2.15
-import QtQuick 2.15
-import QtQuick.Window 2.15
+import QtQml 2.12
+import QtQuick 2.12
+import QtQuick.Window 2.12
 
 import Style 1.0
 import com.nextcloud.desktopclient 1.0 as NC
diff --git a/src/gui/tray/SyncStatus.qml b/src/gui/tray/SyncStatus.qml
index 5964c60db..be357f6d8 100644
--- a/src/gui/tray/SyncStatus.qml
+++ b/src/gui/tray/SyncStatus.qml
@@ -1,6 +1,6 @@
-import QtQuick 2.15
-import QtQuick.Controls 2.15
-import QtQuick.Layouts 1.15
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.12
 
 import Style 1.0
 
diff --git a/src/gui/tray/UnifiedSearchInputContainer.qml b/src/gui/tray/UnifiedSearchInputContainer.qml
index cb5687530..c0b5ec4a0 100644
--- a/src/gui/tray/UnifiedSearchInputContainer.qml
+++ b/src/gui/tray/UnifiedSearchInputContainer.qml
@@ -1,5 +1,5 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import QtGraphicalEffects 1.0
 import Style 1.0
diff --git a/src/gui/tray/UnifiedSearchResultFetchMoreTrigger.qml b/src/gui/tray/UnifiedSearchResultFetchMoreTrigger.qml
index 02ea71f2b..964d33f48 100644
--- a/src/gui/tray/UnifiedSearchResultFetchMoreTrigger.qml
+++ b/src/gui/tray/UnifiedSearchResultFetchMoreTrigger.qml
@@ -1,5 +1,5 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import QtQuick.Layouts 1.2
 import Style 1.0
diff --git a/src/gui/tray/UnifiedSearchResultItem.qml b/src/gui/tray/UnifiedSearchResultItem.qml
index 702b3a331..7e62f694f 100644
--- a/src/gui/tray/UnifiedSearchResultItem.qml
+++ b/src/gui/tray/UnifiedSearchResultItem.qml
@@ -1,4 +1,4 @@
-import QtQml 2.15
+import QtQml 2.12
 import QtQuick 2.9
 import QtQuick.Controls 2.3
 import QtQuick.Layouts 1.2
diff --git a/src/gui/tray/UnifiedSearchResultItemSkeleton.qml b/src/gui/tray/UnifiedSearchResultItemSkeleton.qml
index ddc38d29d..b33e203e1 100644
--- a/src/gui/tray/UnifiedSearchResultItemSkeleton.qml
+++ b/src/gui/tray/UnifiedSearchResultItemSkeleton.qml
@@ -1,5 +1,5 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import QtQuick.Layouts 1.2
 import Style 1.0
 
diff --git a/src/gui/tray/UnifiedSearchResultItemSkeletonContainer.qml b/src/gui/tray/UnifiedSearchResultItemSkeletonContainer.qml
index e128f6860..f5298a925 100644
--- a/src/gui/tray/UnifiedSearchResultItemSkeletonContainer.qml
+++ b/src/gui/tray/UnifiedSearchResultItemSkeletonContainer.qml
@@ -1,5 +1,5 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import Style 1.0
 
diff --git a/src/gui/tray/UnifiedSearchResultListItem.qml b/src/gui/tray/UnifiedSearchResultListItem.qml
index ce942a4d1..69bb3166a 100644
--- a/src/gui/tray/UnifiedSearchResultListItem.qml
+++ b/src/gui/tray/UnifiedSearchResultListItem.qml
@@ -1,5 +1,5 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import Style 1.0
 import com.nextcloud.desktopclient 1.0
diff --git a/src/gui/tray/UnifiedSearchResultNothingFound.qml b/src/gui/tray/UnifiedSearchResultNothingFound.qml
index 67bd8108e..f012ffc5a 100644
--- a/src/gui/tray/UnifiedSearchResultNothingFound.qml
+++ b/src/gui/tray/UnifiedSearchResultNothingFound.qml
@@ -1,5 +1,5 @@
-import QtQml 2.15
-import QtQuick 2.15
+import QtQml 2.12
+import QtQuick 2.12
 import QtQuick.Controls 2.3
 import QtQuick.Layouts 1.2
 import Style 1.0
@@ -7,7 +7,7 @@ import Style 1.0
 ColumnLayout {
     id: unifiedSearchResultNothingFoundContainer
 
-    required property string text
+    property string text
 
     spacing: 8
     anchors.leftMargin: 10
diff --git a/src/gui/tray/UnifiedSearchResultSectionItem.qml b/src/gui/tray/UnifiedSearchResultSectionItem.qml
index 89ad6bf31..04b687a6d 100644
--- a/src/gui/tray/UnifiedSearchResultSectionItem.qml
+++ b/src/gui/tray/UnifiedSearchResultSectionItem.qml
@@ -1,12 +1,12 @@
-import QtQml 2.15
-import QtQuick 2.15
-import QtQuick.Controls 2.15
+import QtQml 2.12
+import QtQuick 2.12
+import QtQuick.Controls 2.12
 import QtQuick.Layouts 1.2
 import Style 1.0
 import com.nextcloud.desktopclient 1.0
 
 Label {
-    required property string section
+    property string section
 
     topPadding: 8
     bottomPadding: 8
diff --git a/src/gui/tray/Window.qml b/src/gui/tray/Window.qml
index 6419e651e..efc4eda8f 100644
--- a/src/gui/tray/Window.qml
+++ b/src/gui/tray/Window.qml
@@ -1,6 +1,6 @@
 import QtQml 2.12
 import QtQml.Models 2.1
-import QtQuick 2.15
+import QtQuick 2.12
 import QtQuick.Window 2.3
 import QtQuick.Controls 2.3
 import QtQuick.Layouts 1.2
@@ -59,6 +59,7 @@ Window {
             accountMenu.close();
             syncStatus.model.load();
         }
+        onNewUserSelected: onNewUserSelected()
     }
 
     Component {
@@ -98,16 +99,18 @@ Window {
             Systray.isOpen = true;
             UserModel.fetchCurrentActivityModel();
         }
+        onShowWindow: onShowWindow()
 
         function onHideWindow() {
             trayWindow.hide();
             Systray.isOpen = false;
         }
+        onHideWindow: onHideWindow()
 
         function onShowFileActivityDialog(objectName, objectId) {
             openFileActivityDialog(objectName, objectId)
         }
-
+        onShowFileActivityDialog: onShowFileActivityDialog(sharePath, localPath)
         function onShowErrorMessageDialog(error) {
             var newErrorDialog = errorMessageDialog.createObject(trayWindow)
             newErrorDialog.text = error
@@ -772,7 +775,7 @@ Window {
 
                 keyNavigationEnabled: true
 
-                reuseItems: true
+                //TODO: enable it when Qt 5.15 is available for every OS reuseItems: true
 
                 Accessible.role: Accessible.List
                 Accessible.name: qsTr("Unified search results list")
@@ -794,8 +797,6 @@ Window {
                     currentFetchMoreInProgressProviderId: unifiedSearchResultsListView.model.currentFetchMoreInProgressProviderId
                     fetchMoreTriggerClicked: unifiedSearchResultsListView.model.fetchMoreTriggerClicked
                     resultClicked: unifiedSearchResultsListView.model.resultClicked
-                    ListView.onPooled: isPooled = true
-                    ListView.onReused: isPooled = false
                 }
 
                 section.property: "providerName"
